python/semantic_kernel/connectors/memory/astradb/astradb_memory_store.py

```diff
@@ -0,0 +1,298 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from logging import Logger
+from typing import List, Optional, Tuple
+
+from numpy import ndarray
+
+from semantic_kernel.connectors.memory.astradb.astra_client import AstraClient
+from semantic_kernel.connectors.memory.astradb.utils import (
+    build_payload,
+    parse_payload,
+)
+from semantic_kernel.memory.memory_record import MemoryRecord
+from semantic_kernel.memory.memory_store_base import MemoryStoreBase
+from semantic_kernel.utils.null_logger import NullLogger
+
+# Limitations set by Pinecone at https://docs.pinecone.io/docs/limits
+MAX_DIMENSIONALITY = 20000
+MAX_UPSERT_BATCH_SIZE = 100
+MAX_QUERY_WITHOUT_METADATA_BATCH_SIZE = 10000
+MAX_QUERY_WITH_METADATA_BATCH_SIZE = 1000
+MAX_FETCH_BATCH_SIZE = 1000
+MAX_DELETE_BATCH_SIZE = 1000
+
+
+class AstraDBMemoryStore(MemoryStoreBase):
+    """A memory store that uses Pinecone as the backend."""
+
+    _logger: Logger
+
+    def __init__(
+        self,
+        astra_application_token: str,
+        astra_id: str,
+        astra_region: str,
+        keyspace_name: str,
+        embedding_dim: int,
+        similarity: str,
+        logger: Optional[Logger] = None,
+    ) -> None:
+        """Initializes a new instance of the AstraDBMemoryStore class.
+
+        Arguments:
+            app_token {str} -- The Astra application token.
+            db_id {str} -- The Astra id of database.
+            regin {str} -- The Astra region
+            keyspace {str} -- The Astra keyspace
+            embedding_dim {int} -- The dimensionality to use for new collections.
+            logger {Optional[Logger]} -- The logger to use. (default: {None})
+        """
+        self._embedding_dim = embedding_dim
+        self._similarity = similarity
+
+        if self._embedding_dim > MAX_DIMENSIONALITY:
+            raise ValueError(
+                f"Dimensionality of {self._embedding_dim} exceeds "
+                + f"the maximum allowed value of {MAX_DIMENSIONALITY}."
+            )
+
+        self._client = AstraClient(
+            astra_id=astra_id, astra_region=astra_region, astra_application_token=astra_application_token, keyspace_name=keyspace_name, embedding_dim=embedding_dim, similarity_function=similarity)
+
+        self._logger = logger or NullLogger()
+
+    def get_collections(self) -> List[str]:
+        """Gets the list of collections.
+
+        Returns:
+            List[str] -- The list of collections.
+        """
+        return self._client.find_collections(False)
```

Comment:
Hi @eavanvalkenburg - our team updated the code so we are now passing all tests with the functions converted to Async

Reply:
Thanks, but the methods in the client themselves should be async, this way it will run new loops many times leading to all sorts of weird behaviours, so it is going async main -> sync here -> async.run(async method in client) which is really not necessary and not the way it should be.

Reply:
Okay- this is now done too @eavanvalkenburg - thanks for the clarification